src/types/resource_types.ts

```diff
@@ -1,21 +1,23 @@
 import type { IrodsResponse, AVUOperation } from './general_types.js'
 
+export type ResourceType =
+    | 'compound'
+    | 'deferred'
+    | 'load_balanced'
+    | 'mockarchive'
+    | 'nonblocking'
+    | 'passthru'
+    | 'random'
+    | 'replication'
+    | 'structfile'
+    | 'univmss'
+    | 'unixfilesystem'
+
```

Comment:
Users can implement new resource types for iRODS. For example, I can write a new resource plugin and call it whatever I want. Is that use-case supported by this? If not, the type should just be a string.

Reply:
It is not, I'll change it to be just a string.